packages/client/src/prax.ts

```diff
@@ -15,82 +17,99 @@ const prax_id = 'lkpmkhpnhknhmibgnmmhdhgdilepfghe';
 const prax_manifest = `chrome-extension://${prax_id}/manifest.json`;
 
-export class PraxNotAvailableError extends Error {}
-export class PraxNotConnectedError extends Error {}
-export class PraxNotInstalledError extends Error {}
-export class PraxManifestError extends Error {}
-
-export const getPraxPort = async () => {
-  const provider = window[PenumbraSymbol]?.[prax_origin];
-  if (!provider) throw new Error('Prax not installed');
-  return provider.connect();
-};
-
-export const requestPraxConnection = async () => {
-  if (window[PenumbraSymbol]?.[prax_origin]?.manifest !== prax_manifest) {
-    throw new PraxManifestError('Incorrect Prax manifest href');
-  }
-  return window[PenumbraSymbol][prax_origin].request();
-};
+export const isPraxAvailable = () => Boolean(window[PenumbraSymbol]?.[prax_origin]);
 
 export const isPraxConnected = () => Boolean(window[PenumbraSymbol]?.[prax_origin]?.isConnected());
 
-export const isPraxConnectedTimeout = (timeout: number) =>
-  new Promise<boolean>(resolve => {
-    if (window[PenumbraSymbol]?.[prax_origin]?.isConnected()) resolve(true);
-
-    const interval = setInterval(() => {
-      if (window[PenumbraSymbol]?.[prax_origin]?.isConnected()) {
-        clearInterval(interval);
-        resolve(true);
-      }
-    }, 100);
-
-    setTimeout(() => {
-      clearInterval(interval);
-      resolve(false);
-    }, timeout);
-  });
-
-export const throwIfPraxNotConnectedTimeout = async (timeout = 500) => {
-  const isConnected = await isPraxConnectedTimeout(timeout);
-  if (!isConnected) throw new PraxNotConnectedError('Prax not connected');
+export const isPraxInstalled = async () => {
+  try {
+    // can check independent of injection presence, as the manifest url is known
+    const { ok } = await fetch(prax_manifest);
+    return ok;
+  } catch {
+    return false;
+  }
 };
 
-export const isPraxAvailable = () => Boolean(window[PenumbraSymbol]?.[prax_origin]);
-
 export const throwIfPraxNotAvailable = () => {
-  if (!isPraxAvailable()) throw new PraxNotAvailableError('Prax not available');
+  if (!isPraxAvailable()) throw new PraxNotAvailableError();
 };
 
 export const throwIfPraxNotConnected = () => {
-  if (!isPraxConnected()) throw new PraxNotConnectedError('Prax not connected');
+  if (!isPraxConnected()) throw new PraxNotConnectedError();
 };
 
-export const getPraxManifest = async () => {
-  const response = await fetch(prax_manifest);
-  return (await response.json()) as JsonValue;
+export const throwIfPraxNotInstalled = async () => {
+  if (!(await isPraxInstalled())) throw new PraxNotInstalledError();
 };
 
-export const isPraxInstalled = () =>
-  getPraxManifest().then(
-    () => true,
-    () => false,
+export const getPraxManifest = async () => {
+  const json = await fetch(prax_manifest).then(
+    ({ ok, json }) => (ok ? json : Promise.reject(new PraxNotInstalledError())),
+    () => Promise.reject(new PraxNotInstalledError()),
   );
+  return (await json()) as JsonValue;
+};
```

Comment:
let's not mix async await and then() syntax here. Also, don't need to do Promise.reject() with async functions. Maybe something like:


```ts
export const getPraxManifest = async (): Promise<JsonValue> => {
  try {
    const response = await fetch(prax_manifest);
    if (!response.ok) {
      throw new PraxNotInstalledError();
    }
    return await response.json() as JsonValue;
  } catch (error) {
    throw new PraxNotInstalledError();
  }
};

```